docs/bokeh/source/docs/user_guide/basic/annotations.rst

```diff
@@ -215,8 +215,20 @@ See |interactive legends| in the user guide for more information and examples.
 Color bars
```

Comment:
How about we first talk about the `construct_color_bar` as the general way to add a color bar, and then say something like "And you also have the option to create a ColorBar object etc."? 
In the Titles and Legends sections that precede the ColotBar section, we also focus on the plotting ploting interface method first!

Reply:
I personally prefer this order, since I think its important to establish that the color bar is driven by a color mapper from the start. I also think the example using `construct_color_bar` coming immediately after the introduction of `construct_color_bar` is natural. But if you would like to push changes to this PR I will demur.